tests/test_make.py

```diff
@@ -309,18 +322,22 @@ class A:
 
     schema = desert.schema_class(A)()
 
-    assert schema.loads('{"x": "1.3"}') == A(decimal.Decimal("1.3"))
+    assert schema.loads('{"x": "1.3"}') == A(decimal.Decimal("1.3"))  # type: ignore[call-arg]
```

Comment:
I often do add them.  I probably was just being lazy here.  The issue is that we use `module` and so the attrs and dataclasses mypy plugins can't tell how to modify the classes we define here.  Basically, the handy dynamicness of Python is getting in the way of the static mypy checking.  Any ideas how to address this instead of ignoring?  I guess maybe we could hint `module.dataclass` to support `a`, `b`, and `c` anyways.  I'll take stab at that.  It's a hack, but maybe less silly than the ignores.

Reply:
There's a basic issue that mypy that it doesn't handle class decorator syntax outside of plugins (https://github.com/python/mypy/issues/3135).  I'm not sure if the change below makes it better since you have to apply the decorator without the decorator syntax, and it's a lie, but this works.  Any opinions on this, or other next step ideas?  At this point I would stick with the ignores.  Maybe a single-point explanation in the `DataclassModule` class would be sufficient?

```patch
diff --git a/tests/test_make.py b/tests/test_make.py
index 8efdcdb..b8c45a0 100644
--- a/tests/test_make.py
+++ b/tests/test_make.py
@@ -18,11 +18,21 @@ import typing_extensions
 import desert
 
 
+@attr.frozen
+class C:
+    # Just a workaround to handle some mypy errors due to not being able to
+    # apply the attrs and dataclass awareness through the DataclassModule
+    # parametrization.  Add any additional attributes that are needed by the
+    # tests.
+    x: object = None
+    y: object = None
+
+
 @attr.frozen(order=False)
 class DataclassModule:
     """Implementation of a dataclass module like attr or dataclasses."""
 
-    dataclass: t.Callable[[type], type]
+    dataclass: t.Callable[[type], t.Type[C]]
     field: t.Callable[..., t.Any] = attr.ib()
     fields: t.Callable[[type], object] = attr.ib()
 
@@ -117,13 +127,14 @@ _assert_dump_load = fixture_from_dict(
 def test_simple(module: DataclassModule) -> None:
     """Load dict into a dataclass instance."""
 
-    @module.dataclass
-    class A:
+    class BareA:
         x: int
 
+    A = module.dataclass(BareA)
+
     data = desert.schema_class(A)().load(data={"x": 5})
 
-    assert data == A(x=5)  # type: ignore[call-arg]
+    assert data == A(x=5)
```